### PR TITLE
Consume system transactions in StateComputer

### PR DESCRIPTION
**Pull request by zjma**
### Description

`StateComputer` should consume system transactions in given block when `schedule_compute()/commit()`ing.

### Test Plan
New UTs

### CODE REVIEW

File: consensus/consensus-types/src/block_data.rs [hunk 1]
```diff
@@ -232,6 +232,18 @@ impl BlockData {
         }
     }
 
+    pub fn dummy_with_system_txns(sys_txns: Vec<SystemTransaction>) -> Self {
```
[the remark this answers is not in the document]
Comment (zjma) [reply consensus/consensus-types/src/block_data.rs:235]: tried, but my `#[tokio::test]` can no longer find it

[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/consensus-types/src/block_data.rs:235]: need to define the test file with cfg(test) too?

File: types/src/block_executor/partitioner.rs [hunk 2]
```diff
@@ -545,6 +545,7 @@ impl PartitionedTransactions {
 }
 
 // Represents the transactions in a block that are ready to be executed.
+#[derive(Clone)]
```
[the remark this answers is not in the document]
Comment (zjma) [reply types/src/block_executor/partitioner.rs:548]: [here](https://github.com/aptos-labs/aptos-core/blob/df08e539468cc253797510fdf8c5a12f2f0644d8/consensus/src/state_computer_tests.rs#L171-L174) the test somehow needs it.

as long as production code doesn't use it, it's fine?

File: types/src/ledger_info.rs [hunk 3]
```diff
@@ -157,6 +164,10 @@ impl LedgerInfoWithSignatures {
         LedgerInfoWithSignatures::V0(LedgerInfoWithV0::new(ledger_info, signatures))
     }
 
+    pub fn dummy() -> Self {
```
[the remark this answers is not in the document]
Comment (zjma) [reply types/src/ledger_info.rs]: deleted, it seems `new()` is enough.

File: types/src/validator_verifier.rs [hunk 4]
```diff
@@ -121,6 +121,16 @@ impl<'de> Deserialize<'de> for ValidatorVerifier {
 }
 
 impl ValidatorVerifier {
+    #[cfg(test)]
+    pub fn dummy() -> Self {
```
[the remark this answers is not in the document]
Comment (zjma) [reply types/src/validator_verifier.rs]: deleted, no longer needed.

File: execution/executor-types/src/lib.rs [hunk 5]
```diff
@@ -316,7 +316,7 @@ pub struct StateComputeResult {
     /// The compute status (success/failure) of the given payload. The specific details are opaque
     /// for StateMachineReplication, which is merely passing it between StateComputer and
     /// PayloadClient.
-    compute_status: Vec<TransactionStatus>,
+    pub compute_status: Vec<TransactionStatus>,
```
[the remark this answers is not in the document]
Comment (zjma) [reply execution/executor-types/src/lib.rs]: fixed

File: consensus/src/state_computer.rs [hunk 6]
```diff
@@ -133,14 +133,15 @@ impl StateComputer for ExecutionProxy {
         let txn_deduper = self.transaction_deduper.lock().as_ref().unwrap().clone();
         let txn_shuffler = self.transaction_shuffler.lock().as_ref().unwrap().clone();
         let txn_notifier = self.txn_notifier.clone();
-        let txns = match payload_manager.get_transactions(block).await {
+        let sys_txns = block.sys_txns().map_or(vec![], Vec::clone);
```
[the remark this answers is not in the document]
Comment (zjma) [reply consensus/src/state_computer.rs]: fixed

File: consensus/consensus-types/src/vote_data.rs [hunk 7]
```diff
@@ -39,6 +39,13 @@ impl VoteData {
         Self { proposed, parent }
     }
 
+    pub fn dummy() -> Self {
```
[the remark this answers is not in the document]
Comment (zjma) [reply consensus/consensus-types/src/vote_data.rs:43]: fixed

File: consensus/consensus-types/src/block_data.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (zjma) [reply consensus/consensus-types/src/block_data.rs:236]: fixed

File: consensus/consensus-types/src/block.rs [hunk 8]
```diff
@@ -428,6 +430,7 @@ impl Block {
             once(Transaction::BlockMetadata(
                 self.new_block_metadata(validators),
             ))
+            .chain(sys_txns.into_iter().map(Transaction::SystemTransaction))
```
[the remark this answers is not in the document]
Comment (zjma) [reply consensus/consensus-types/src/block.rs]: fixed

File: types/src/block_executor/partitioner.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (zekun000) [reply types/src/block_executor/partitioner.rs:548]: we can cfg(test) the clone as well, something like this #[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]

File: consensus/consensus-types/src/block.rs [hunk 9]
```diff
@@ -411,26 +411,25 @@ impl Block {
...
         } else {
             // Before the per-block gas limit change, StateCheckpoint txn
             // is inserted here for compatibility.
-            once(Transaction::BlockMetadata(
-                self.new_block_metadata(validators),
-            ))
-            .chain(txns.into_iter().map(Transaction::UserTransaction))
-            .chain(once(Transaction::StateCheckpoint(self.id)))
-            .collect()
+            txns.chain(once(Transaction::StateCheckpoint(self.id)))
```
Comment (zekun000) [inline consensus/consensus-types/src/block.rs:431]: I think we can just write like this?
```
if !is_block_gas_limit {
  txns.chain(..);
}
txns.collect()
```

**Review by zekun000 — APPROVED** [review]

File: consensus/consensus-types/src/block.rs (hunk 9 above)
Comment (zjma) [reply consensus/consensus-types/src/block.rs:431]: `A.chain(B)` returns something of type `Chain<A, B>`, so it's not possible to have an optional step in this pattern I think...

**Review by danielxiangzl — APPROVED** [review]
LGTM

**Pull request merged**